core/app/models/spree/promotion.rb

```diff
@@ -120,7 +120,12 @@ def deactivate(payload)
     def eligible?(promotable)
       return false if expired? || usage_limit_exceeded?(promotable) || blacklisted?(promotable)
 
-      !!eligible_rules(promotable, {})
+      case promotable
+      when Spree::LineItem
+        !!eligible_rules(promotable, {}) && !!eligible_rules(promotable.order, {})
+      else
+        !!eligible_rules(promotable, {})
+      end
     end
 
```

Comment:
let's use early return here. Using `case` when we have only one option looks redundant.

Reply:
```suggestion
def eligible?(promotable)
    return false if expired? || usage_limit_exceeded?(promotable) || blacklisted?(promotable)
    return !!eligible_rules(promotable, {}) && !!eligible_rules(promotable.order, {}) if promotable == Spree::LineItem
    
    !!eligible_rules(promotable, {})
end
```

Reply:
That makes sense. I'll rewrite this.